controllers/admin/AdminImportController.php

```diff
@@ -1596,26 +1596,31 @@ protected function productImportAccessories($offset, $limit, &$crossStepsVariabl
 
     protected function productImportOne($info, $default_language_id, $id_lang, $force_ids, $regenerate, $shop_is_feature_active, $shop_ids, $match_ref, &$accessories, $validateOnly = false)
     {
-        if ($force_ids && isset($info['id']) && (int)$info['id']) {
-            $product = new Product((int)$info['id']);
-        } elseif ($match_ref && array_key_exists('reference', $info)) {
-            $datas = Db::getInstance()->getRow('
-					SELECT p.`id_product`
-					FROM `'._DB_PREFIX_.'product` p
-					'.Shop::addSqlAssociation('product', 'p').'
-					WHERE p.`reference` = "'.pSQL($info['reference']).'"
-				', false);
-            if (isset($datas['id_product']) && $datas['id_product']) {
-                $product = new Product((int)$datas['id_product']);
+        $id_product = null;
+        // Use product reference as key
+        if ($match_ref && array_key_exists('reference', $info)) {
+            $idReference = (int) Db::getInstance()->getValue('
+                                    SELECT p.`id_product`
+                                    FROM `' . _DB_PREFIX_ . 'product` p
+                                    ' . Shop::addSqlAssociation('product', 'p') . '
+                                    WHERE p.`reference` = "' . pSQL($info['reference']) . '"
+                                ', false);
+            if ($idReference) {
+                $id_product = $idReference;
+            }
+        }
+
+        // Force all ID numbers, overrides option Use product reference as key
+        if (array_key_exists('id', $info) && (int) $info['id'] && Product::existsInDatabase((int) $info['id'], 'product')) {
```

Comment:
With the previous comments, this part should not be useful anymore and can be removed.